widelands 1.1-rc1: compilation error: use of undeclared identifier 'PATH_MAX'

**Describe the bug**
When compiling widelands 1.1-rc1 on OpenMandriva Cooker I see build error:

```
/builddir/build/BUILD/widelands-1.1-rc1/src/io/filesystem/filesystem.cc:183:11: error: use of undeclared identifier 'PATH_MAX'
DEBUG util.py:446:          char cwd[PATH_MAX + 1];
DEBUG util.py:446:                   ^
DEBUG util.py:446:  /builddir/build/BUILD/widelands-1.1-rc1/src/io/filesystem/filesystem.cc:188:35: error: use of undeclared identifier 'PATH_MAX'
DEBUG util.py:446:          char* const result = getcwd(cwd, PATH_MAX);
DEBUG util.py:446:                                           ^
DEBUG util.py:446:  2 errors generated.
```

Same error on Clang 15.0.2 and GCC compiler.

Full build log: https://file-store.openmandriva.org/api/v1/file_stores/9e9dcb617964e9d37469c17b9007b6fb930fab0d.log?show=true

**To Reproduce**
Steps to reproduce the behavior:
1. Compile from source widelands 1.1-rc1
2. See error

**Expected behavior**
Compilation ends without errors


**Version:**
 - OS: [e.g. OpenMandriva Cooker]
 - Widelands Version: 1.1-rc1]

--- a/src/io/filesystem/filesystem.cc
+++ b/src/io/filesystem/filesystem.cc
@@ -59,6 +59,10 @@
 #define PATH_MAX MAX_PATH
 #endif
 
+#ifndef PATH_MAX
+#define PATH_MAX 0x1000000
+#endif
+
 namespace {
 /// A class that makes iteration over filename_?.* templates easy. It is much faster than using
 /// regex.

--- a/src/io/filesystem/filesystem.cc
+++ b/src/io/filesystem/filesystem.cc
@@ -59,8 +59,13 @@
 #define PATH_MAX MAX_PATH
 #endif
 
+/* Quickfix for bug https://github.com/widelands/widelands/issues/5614:
+ * Most systems specify PATH_MAX to be the maximum number of characters in a file path.
+ * Systems without a limit (or which don't care about standards) may neglect to define this symbol.
+ * On such systems, simply use an arbitrary value that is high enough for some very long paths.
+ */
 #ifndef PATH_MAX
-#define PATH_MAX 0x1000000
+#define PATH_MAX 0x10000
 #endif
 
 namespace {

